platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementProjectScopeServices.java

```diff
@@ -84,14 +85,26 @@ DependencyFactoryInternal createDependencyFactory(
         return new DefaultDependencyFactory(
                 instantiator,
                 DependencyNotationParser.create(instantiator, factory, classPathRegistry, fileCollectionFactory, runtimeShadedJarFactory, currentGradleInstallation, stringInterner),
-                DependencyConstraintNotationParser.parser(instantiator, factory, stringInterner, attributesFactory),
                 new ClientModuleNotationParserFactory(instantiator, stringInterner).create(),
                 capabilityNotationParser,
                 objectFactory,
                 projectDependencyFactory,
                 attributesFactory);
     }
 
+    DependencyConstraintFactoryInternal createDependencyConstraintFactory(
```

Comment:
Why is this duplicated here and in `DependencyManagementBuildScopeServices`?

Reply:
Technically, we have a different `DefaultProjectDependencyFactory` here than in build scope, in order to be able to resolve tasks in the project context. However, I don't think there is any reason that should matter for constraints, so I'll remove it.